sumpy/kernel.py

```diff
@@ -519,6 +519,10 @@ def __init__(self, dim, helmholtz_k_name="k",
             scaling = var("I")/4
         elif dim == 3:
             r = pymbolic_real_norm_2(make_sym_vector("d", dim))
+            if allow_evanescent:
+                expr = var("exp")(var("I")*k*r)/r
+            else:
+                expr = (var("cos")(k*r) + var("I")*var("sin")(k*r))/r
             expr = var("exp")(var("I")*k*r)/r
```

Comment:
```suggestion
```
? Seems like `expr` is just overwritten by this line?

Reply:
Oops. Thanks for catching.